crit-go/magic-gen/magic-gen.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"github.com/thoas/go-funk"
+	"os"
+	"reflect"
+	"strconv"
+	"strings"
+)
+
+func main() {
+	// checks argument correctness
+	if len(os.Args) != 3 {
+		fmt.Println("Usage: magic_gen.go path/to/magic.h path/to/magic.json")
```

Comment:
I think the resulting binary is called `magic-gen` and not  `magic_gen.go`. Adapt the help output.